src/main/features/response/routes/full-admission/payment-plan.ts

```diff
@@ -1,26 +1,45 @@
 import * as express from 'express'
 
+import * as _ from 'lodash'
+
 import { Paths, FullAdmissionPaths } from 'response/paths'
 
 import { ErrorHandling } from 'shared/errorHandling'
 import { Form } from 'forms/form'
 import { DraftService } from 'services/draftService'
 import { User } from 'idam/user'
-import { PaidAmount } from 'ccj/form/models/paidAmount'
 import { DefendantPaymentPlan } from 'response/form/models/defendantPaymentPlan'
 import { FormValidator } from 'forms/validation/formValidator'
 import { FeatureToggleGuard } from 'guards/featureToggleGuard'
 import { ResponseDraft } from 'response/draft/responseDraft'
 import { Draft } from '@hmcts/draft-store-client'
 import { Claim } from 'claims/models/claim'
+import { createPaymentPlan } from 'common/paymentPlan'
+import { PaymentSchedule } from 'features/ccj/form/models/paymentSchedule'
+
+function mapFrequencyInWeeks (frequency: PaymentSchedule): number {
+  switch (frequency) {
+    case PaymentSchedule.EACH_WEEK:
+      return 1
+    case PaymentSchedule.EVERY_TWO_WEEKS:
+      return 2
+    case PaymentSchedule.EVERY_MONTH:
+      return 4
+    default:
+      return undefined
+  }
+}
 
-function renderView (form: Form<PaidAmount>, res: express.Response): void {
+function renderView (form: Form<DefendantPaymentPlan>, res: express.Response): void {
   const claim: Claim = res.locals.claim
   const draft: Draft<ResponseDraft> = res.locals.responseDraft
   const alreadyPaid: number = draft.document.paidAmount.amount || 0
 
   res.render(FullAdmissionPaths.paymentPlanPage.associatedView, {
     form: form,
+    paymentLength: calculatePaymentPlanLength(form.model),
+    monthlyIncome: _.get(draft, 'document.statementOfMeans.monthlyIncome', 0),
+    monthlyExpenses: _.get(draft, 'document.statementOfMeans.monthlyExpenses', 0),
     remainingAmount: claim.totalAmountTillToday - alreadyPaid
```

Comment:
I would suggest double checking with PO whether we need to include interest till today or just interest till date of issue.

Reply:
there is no mention of interest in this screen in prototype, could you please clarify what you mean? Do you mean use interest when calculating remaining amount? I assumed remaining amount is amount after all calculations..